builtin/logical/pki/path_revoke.go

```diff
@@ -418,12 +417,32 @@ func (b *backend) pathRevokeWrite(ctx context.Context, req *logical.Request, dat
 
 	// We store and identify by lowercase colon-separated hex, but other
 	// utilities use dashes and/or uppercase, so normalize
-	colonSerial := denormalizeSerial(serial)
+	hyphenSerial := normalizeSerial(serial)
 	b.revokeStorageLock.Lock()
 	defer b.revokeStorageLock.Unlock()
 
-	sc := b.makeStorageContext(ctx, req.Storage)
-	return revokeCert(sc, colonSerial, false)
+	certEntry, err := fetchCertBySerial(sc, "certs/", hyphenSerial)
+	if err != nil {
+		switch err.(type) {
+		case errutil.UserError:
+			return logical.ErrorResponse(err.Error()), nil
+		default:
+			return nil, err
+		}
+	}
+	if certEntry == nil {
+		return logical.ErrorResponse(fmt.Sprintf("certificate with serial %s not found", denormalizeSerial(serial))), nil
+	}
+
+	cert, err := x509.ParseCertificate(certEntry.Value)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing certificate: %w", err)
+	}
+	if cert == nil {
+		return nil, fmt.Errorf("got a nil certificate")
```

Comment:
I don't think this can happen fwiw.

Reply:
Yeah, I figured, but I was trying to keep the code as close as possible to the original for the refactoring.

Reply:
So it looks like I removed this already in [c32d068](https://github.com/hashicorp/vault/pull/18685/commits/c32d068444970930d7317b677f57e99896a6d532) so, so much for me not trying to keep the code as close as possible...